coriolisclient/tests/cli/test_shell.py

```diff
@@ -0,0 +1,476 @@
+                    create_keystone_session_args,
+                    auth_type=data["auth_type"]
+                )
+                mock_Client.assert_called_once_with(
+                    session=mock_create_keystone_session.return_value,
+                    **call_args
+                )
+        else:
+            self.assertRaises(
+                exceptions.CoriolisException,
+                self.coriolis.create_client,
+                args
+            )
+
+    def test_build_kwargs_based_on_version(self):
+        args = CustomMock()
+        args.os_project_id = mock.sentinel.os_project_id
+        args.os_tenant_name = mock.sentinel.os_tenant_name
+
+        result = self.coriolis.build_kwargs_based_on_version(args)
+
+        self.assertEqual(
+            {'project_id': mock.sentinel.os_project_id},
+            result
+        )
+
+        api_version = '2'
+
+        result = self.coriolis.build_kwargs_based_on_version(args, api_version)
+
+        self.assertEqual(
+            {'tenant_name': mock.sentinel.os_tenant_name},
+            result
+        )
+
+    def test_get_endpoint_filter_kwargs(
+        self
+    ):
+        args = mock.Mock()
+        args.interface = mock.sentinel.interface
+        args.service_type = mock.sentinel.service_type
+        args.service_name = mock.sentinel.service_name
+        args.coriolis_api_version = mock.sentinel.coriolis_api_version
+        args.region_name = mock.sentinel.region_name
+        expected_result = {
+            "interface": mock.sentinel.interface,
+            "service_type": mock.sentinel.service_type,
+            "service_name": mock.sentinel.service_name,
+            "region_name": mock.sentinel.region_name,
+            "version": mock.sentinel.coriolis_api_version,
+        }
+
+        result = self.coriolis._get_endpoint_filter_kwargs(args)
+
+        self.assertEqual(
+            expected_result,
+            result
+        )
+
+    @mock.patch.object(shell.Coriolis, '_env')
+    @mock.patch.object(loading, 'register_session_argparse_arguments')
+    @mock.patch.object(app.App, 'build_option_parser')
+    def test_build_option_parser(
+        self,
+        mock_build_option_parser,
+        mock_register_session_argparse_arguments,
+        *_
+    ):
+        result = self.coriolis.build_option_parser(
+            mock.sentinel.description, mock.sentinel.version)
+
+        self.assertEqual(
+            mock_build_option_parser.return_value,
+            result
+        )
+        mock_build_option_parser.assert_called_once_with(
+            mock.sentinel.description, mock.sentinel.version, None)
+        mock_register_session_argparse_arguments.assert_called_once_with(
+            mock_build_option_parser.return_value)
+
+    def test_env(self):
+        result = self.coriolis._env(
+            "HOME", mock.sentinel.default)
+
+        self.assertEqual(
+            os.environ.get("HOME"),
```

Comment:
You're basically mimicking the method here. Please instead set a random env value and check that the method returns it. (using `os.environ`) 
```python3
os.environ["TEST_ENV"] = "test_value"
```